### PR TITLE
feat: add github actions workflows

### PR DESCRIPTION
**Pull request by kneal**
**This PR contains _(Pulled files from existing projects)_:**

* Standard CI pipeline workflows 
* Changelog templates required for changelog workflow
* Add standard GolangCI lint rules
* Add standard CodeCov file

**Note: This build will fail because it does not have the make file that is in the other PR**

### CODE REVIEW

File: .chglog/config.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+style: github
+template: CHANGELOG.tpl.md
+info:
+  title: CHANGELOG
+  repository_url: https://github.com/go-vela/vela-terraform
```
Comment (jbrockopp) [inline .chglog/config.yml:5]: I think we'll want to update this to point at this repo 👍 

```suggestion
  repository_url: https://github.com/go-vela/secret-vault
```

**Comment by kneal** [comment]
@wass3r it was in the other PR:
https://github.com/go-vela/secret-vault/pull/2/files#diff-b67911656ef5d18c4ae36cb6741b7965R5

I'm trying to figure out how to update this branch. I don't see it in the PR box for "update branch"

**Comment by kneal** [comment]
@wass3r think we got it now

**Comment by kneal** [comment]
@wass3r this look good now?

**Pull request merged**